### PR TITLE
Use automatic naming of the RDS

### PR DESCRIPTION
**Pull request by sevenmachines**
This removes the use of the RDS identifier setting.
- Pass template to config:rds method
- Randomise a DBInstanceIdentifierName
- Update tests to use template passed to rds()
- Changed 2 test comparisons to standard expected_data, actual_data order
- Drop mention of identifier from sample project

(Closes ministryofjustice/bootstrap-cfn#137)

### CODE REVIEW

**Comment by mattmb** [comment]
Hmmm, I think it's the identifier that's most problematic?

Maybe we should have random DBName too. But it's the identifier that has to be unique.

File: bootstrap_cfn/config.py [hunk 1]
```diff
@@ -356,6 +357,7 @@ def rds(self):
             AutoMinorVersionUpgrade=False,
             VPCSecurityGroups=[GetAtt(database_sg, "GroupId")],
             DBSubnetGroupName=Ref(rds_subnet_group),
+            DBInstanceIdentifier=identifier,
```
Comment (ashb) [inline bootstrap_cfn/config.py]: I don't think we should need to specify this. From http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-rds-database-instance.html#cfn-rds-dbinstance-dbinstanceidentifier

> **DBInstanceIdentifier**
> 
>    A name for the DB instance. If you don't specify a name, AWS CloudFormation generates a unique physical ID and uses that ID for the DB instance. For more information, see Name Type.
> 
>    _Required: No_

File: tests/tests.py [hunk 2]
```diff
@@ -564,7 +592,7 @@ def test_process_with_vpc_config(self):
         wanted = ["dbhost", "dbport"]
         output_names = cfn_template['Outputs'].keys()
         output_names.sort()
-        compare(output_names, wanted)
+        compare(wanted, output_names)
```
Comment (ashb) [inline tests/tests.py:681]: I'm not sure if there's a proper order for wanted vs expected (the docs just say one and two I think) but why did we swap these parameters around?

File: bootstrap_cfn/config.py [hunk 3]
```diff
@@ -312,7 +310,6 @@ def rds(self):
             'backup-retention-period': 'BackupRetentionPeriod',
             'db-master-username': 'MasterUsername',
             'db-master-password': 'MasterUserPassword',
-            'identifier': 'DBInstanceIdentifier',
```
Comment (ashb) [inline bootstrap_cfn/config.py:332]: To make back-compat/upgrade easy could we make this an optional field?

Comment (sevenmachines) [inline bootstrap_cfn/config.py:332]: will do

File: bootstrap_cfn/config.py (hunk 1 above)
Comment (sevenmachines) [inline bootstrap_cfn/config.py]: Can do this as part of the above change too

File: tests/tests.py (hunk 2 above)
Comment (sevenmachines) [inline tests/tests.py:681]: Not proper so much as consistent, mosts of the tests run an expected/actual comparison order, best to keep them all the same if possible

**Pull request merged**